db/blip_sync_context.go

```diff
@@ -606,52 +611,68 @@ func (bsc *BlipSyncContext) sendNoRev(sender *blip.Sender, docID, revID string,
 }
 
 // Pushes a revision body to the client
-func (bsc *BlipSyncContext) sendRevision(sender *blip.Sender, docID, revID string, seq SequenceID, knownRevs map[string]bool, maxHistory int, handleChangesResponseCollection *DatabaseCollectionWithUser, collectionIdx *int) error {
-	rev, err := handleChangesResponseCollection.GetRev(bsc.loggingCtx, docID, revID, true, nil)
+func (bsc *BlipSyncContext) sendRevision(sender *blip.Sender, docID, rev string, seq SequenceID, knownRevs map[string]bool, maxHistory int, handleChangesResponseCollection *DatabaseCollectionWithUser, collectionIdx *int) error {
+	var err error
+	var docRev DocumentRevision
+	if bsc.activeCBMobileSubprotocol <= CBMobileReplicationV3 {
+		docRev, err = handleChangesResponseCollection.GetRev(bsc.loggingCtx, docID, rev, true, nil)
+	} else {
+		// extract cv string rev representation
+		version, vrsErr := CreateVersionFromString(rev)
```

Comment:
CreateVersionFromString is a relatively expensive operation (base64 decoding and hex cas conversion).  It's particularly true here, since these are coming from changeArray (which we just constructed when we sent a batch of changes).  
More broadly, I feel like we should discuss whether we can have the majority of our code work with the formatted (base64 and string cas) in memory, to avoid constant conversion back and forth.